include/control_filters/low_pass_filter.hpp

```diff
@@ -0,0 +1,217 @@
+{
+}
+
+template <typename T>
+LowPassFilter<T>::~LowPassFilter()
+{
+}
+
+template <typename T>
+bool LowPassFilter<T>::configure()
+{
+    configure("");
+    return true;
+}
+
+template <typename T>
+bool LowPassFilter<T>::configure(const std::string& ns)
+{
+    if (!initialized_)
+    {
+        RCLCPP_INFO(logger_, "Node is not initialized... call setNode()");
+        return false;
+    }
+
+    parameter_namespace_ = ns;
+    
+    if (!updateParameters())
+    {
+        return false;
+    }
+    
+    configured_ = true; 
+
+    RCLCPP_INFO(logger_,
+                "Low Pass Filter Params: sampling frequency: %f; damping frequency: %f; damping intensity :%f; "
+                "divider :%d",
+                sampling_frequency_, damping_frequency_, damping_intensity_, divider_);
+
+    // Initialize storage Vectors
+    filtered_value = filtered_old_value = old_value = 0;
+    for (unsigned int i = 0; i < 6; i++)
+    {
+        msg_filtered(i) = msg_filtered_old(i) = msg_old(i) = 0;
+    }
+
+    return true;
+}
+
+template <>
+inline bool LowPassFilter<geometry_msgs::msg::WrenchStamped>::update(const geometry_msgs::msg::WrenchStamped& data_in,
+                                                                     geometry_msgs::msg::WrenchStamped& data_out)
+{
+    if (!configured_)
+    {
+        RCLCPP_ERROR(logger_, "Filter is not configured");
+        return false;
+    }
+
+    if (params_updated_)
+    {
+        updateParameters();
+    }
+
+    // IIR Filter
+    msg_filtered = b1 * msg_old + a1 * msg_filtered_old;
+    msg_filtered_old = msg_filtered;
+
+    // TODO use wrenchMsgToEigen
+    msg_old[0] = data_in.wrench.force.x;
+    msg_old[1] = data_in.wrench.force.y;
+    msg_old[2] = data_in.wrench.force.z;
+    msg_old[3] = data_in.wrench.torque.x;
+    msg_old[4] = data_in.wrench.torque.y;
+    msg_old[5] = data_in.wrench.torque.z;
+
+    data_out.wrench.force.x = msg_filtered[0];
+    data_out.wrench.force.y = msg_filtered[1];
+    data_out.wrench.force.z = msg_filtered[2];
+    data_out.wrench.torque.x = msg_filtered[3];
+    data_out.wrench.torque.y = msg_filtered[4];
+    data_out.wrench.torque.z = msg_filtered[5];
+    return true;
+}
+
+template <typename T>
+bool LowPassFilter<T>::update(const T& data_in, T& data_out)
+{
+    if (!configured_)
+    {
+        RCLCPP_ERROR(logger_, "Filter is not configured");
+        return false;
+    }
+
+    if (params_updated_)
+    {
+        if (!updateParameters()) 
+        {
+            RCLCPP_ERROR(logger_, "Unable to update as not all parameters are set");
+            return false;
+        }
+    }
+
+    data_out = b1 * old_value + a1 * filtered_old_value;
+    filtered_old_value = data_out;
+    old_value = data_in;
+
+    return true;
+}
+
+template <typename T>
+bool LowPassFilter<T>::updateParameters()
+{
+    
+    if (!filters::FilterBase<T>::getParam(parameter_namespace_ + "sampling_frequency", sampling_frequency_)) {
```

Comment:
is this `parameter_namespace_` necessary? Shouldn't this work git `getParam` automatically? How is this done in the filters in `ros2/filters` package?